test/middlewares_test.exs

```diff
@@ -0,0 +1,179 @@
+defmodule MiddlewaresTest do
+  use ExUnit.Case
+
+  alias Exq.Middlewares.Server, as: Middlewares
+  alias Exq.Worker.Server, as: Worker
+
+  defmodule NoArgWorker do
+    def perform do
+    end
+  end
+
+  defmodule MissingMethodWorker do
+  end
+
+  defmodule MyMiddleware do
+    @behaviour Exq.Middlewares.Behaviour
+
+    import Exq.Middlewares.Pipeline
+
+    def before_work(pipeline) do
+      send :middlewarestest, :before_work
+      pipeline = assign(pipeline, :process_info, 1)
```

Comment:
don't need pipeline variable definition (also repeated in a few places above/below)